### PR TITLE
#76 adsl vignette

### PR DESCRIPTION
**Pull request by vikrams95**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/main/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/main/articles/unit_test_guidance.html#writing-unit-tests-in-admiral)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/main/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page.
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue so that it closes after successful merging.
- [ ] Address all merge conflicts and resolve appropriately.
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/adsl.Rmd [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+  comment = "#>"
+)
+library(admiraldev)
+library(admiral)
+library(admiralvaccine)
+```
+
+# Introduction
+
+This article describes creating an `ADSL` ADaM. Examples are currently
```
Comment (bms63) [inline vignettes/adsl.Rmd:23]:
```suggestion
This article describes creating an `ADSL` ADaM specific to Vaccines. Examples are currently 
```

File: vignettes/adsl.Rmd [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+The mapping of the treatment variables is left to the ADaM programmer. An
+example mapping for a study without periods may be:
+
+```{r eval=TRUE}
+adsl <- dm %>%
+  mutate(TRT01P = ARM, TRT01A = ACTARM)
+```
+
+For studies with periods see the ["Visit and Period Variables"
+vignette](visits_periods.html#treatment_adsl).
```
Comment (bms63) [inline vignettes/adsl.Rmd:94]: We should say something about this being in core `admiral`

File: vignettes/adsl.Rmd [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  adsl,
+  display_vars = exprs(USUBJID, TRTSDT, ARM, ACTARM, SAFFL)
+)
+```
+
+## Derive vaccination date Variables  {#vax_date}
+
+In this step, we will create a vaccination date variables from `EX` domain. The function `derive_vars_vaxdt` returns the variables `VAX01DT`,`VAX02DT`... added to the `adsl` dataset based on number of vaccinations.
+
+If there are multiple vaccinations for a visit per subject,warning will be
```
Comment (bms63) [inline vignettes/adsl.Rmd:198]:
```suggestion
If there are multiple vaccinations for a visit per subject, warning will be
```

File: vignettes/adsl.Rmd [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+provided and only first observation will be filtered based on the variable
+order specified on the `order` argument. In this case, user need to select
+the `by_vars` appropriately.
```
Comment (bms63) [inline vignettes/adsl.Rmd:201]:
```suggestion
order specified on the `order` argument. In this case, a user needs to select
the `by_vars` appropriately.
```

File: vignettes/adsl.Rmd [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+NOTE: All these packages are in the experimental phase, but the vision is to
+have them associated with an End to End pipeline under the umbrella of the
+[pharmaverse](https://github.com/pharmaverse). An example of applying metadata 
+and perform associated checks can be found at the [pharmaverse E2E example](https://examples.pharmaverse.org/data/adsl/).
+
+# Example Script 
+
+ADaM | Sample Code
+---- | --------------
+ADSL | [ad_adsl.R](https://github.com/pharmaverse/admiral/blob/main/inst/templates/ad_adsl.R){target="_blank"}
```
Comment (bms63) [inline vignettes/adsl.Rmd:258]: this will need to point the admiralvaccine template eventually

File: vignettes/adsl.Rmd (hunk 1 above)
Comment (vikrams95) [reply vignettes/adsl.Rmd:23]: Fixed it.

File: vignettes/adsl.Rmd (hunk 2 above)
Comment (vikrams95) [reply vignettes/adsl.Rmd:94]: Added.

File: vignettes/adsl.Rmd (hunk 3 above)
Comment (vikrams95) [reply vignettes/adsl.Rmd:198]: Done

File: vignettes/adsl.Rmd (hunk 4 above)
Comment (vikrams95) [reply vignettes/adsl.Rmd:201]: Done.

File: vignettes/adsl.Rmd (hunk 5 above)
Comment (vikrams95) [reply vignettes/adsl.Rmd:258]: We should add the link once it is pushed to main branch right?, we need to rename the template as well. We will update the link later once we are good?

Comment (bms63) [reply vignettes/adsl.Rmd:260]: that is correct - can you make an issue so we don't forgot to make this update.

Comment (vikrams95) [reply vignettes/adsl.Rmd:260]: Sure Ben. Thanks!!

**Review by arjoon-r — APPROVED** [review]

**Pull request merged**